### DISCUSSION TITLE
Deployment Best Practices

### DISCUSSION THREAD

**Discussion by bharat715** [globaleaks/globaleaks-whistleblowing-software#4524] [Q&A]
Is are there any best practice recommendations around deployment architecture document available? 
Development, test, demo and prod environment?
We are considering to apply recent GlobaLeaks upgrade so wanted to know if there any best practices recommended to follow.
Are there any known issues like loos of our installation specific configuration that we need to consider and check after we apply upgrade?

**Comment by evilaliv3** [comment]
Thank you for the question @bharat715 !

At the current moment all the documentation available is published on https://docs.globaleaks.org

If you would like to elaborate a set of questions i would be happy to provide you answers to them so that maybe we could try to cope with any missing detail and enrich the documentation. Would it work for you? 

To summarize:
- globaleaks is not a mission critical software so that an extensive separation from preproduction to production is typically not required and costly
- a production environment require just a small server: a 2 core system with 2 GB of RAM is typically more than enough. You would just required disk
- preproduction do not need to be typically always  ON; you could have a vm where to reproduce the setup and test the update safely and update in production when satisfied.
- to retest you just need to: setup a server like the original one, load the directory with the actual data, and update to see if the update process works and you are satisfied of the result or you need to retest some new configurations.

I welcome your questions and lets start from here!
